### ISSUE TITLE
not possible to continue a chat session

### ISSUE THREAD

**Issue by KorbinianMossandl** [KohlsAdrian/zendesk2#36]
if you don't end the chat session via `_zendeskChat.endChat()` because you want to let a user come back to the same chat session it is not possible because 
```
_zendeskChat.providersStream?.listen((providerModel) {
  setState(() {
    _providerModel = providerModel;
  });
});
```
will never set the _providerModel. Only if someone sends a message the listen method will be triggered and the `_providerModel` will be set and show all the history. 
So i think we need a change where subscribing to the stream always updates `_providerModel` if there already is a chat history.

**Comment by rupas385** [comment]
@KohlsAdrian same problem here. 

The example still with circular loading until new message sent from the agent or agent start typing in the meantime the app only show a circular loading.

**Comment by rupas385** [comment]
I notice this problem only affect on Android, on iOS works well.

For reproduce:

1.- Connect to chat (works well)
2.- Sent a message (works well)
3.- tap on back button and call dispose and disconnect
4.- return to chat and see that only see the circular indicator until agent start typing or sent a message

**Comment by KorbinianMossandl** [comment]
you are right, it seems to only affect android. so imo we need a fix where the existing model is always send to a subscriber the moment he subscribes.
My temporary fix for the moment is to  save the model and reload it if the chat was not ended. but thats just a workaround imo.

**Comment by rupas385** [comment]
Fro the moment my solution is instead of compare if provider model is null I compare if the connection status is connected, if yes then show the chat screen, then when the user send a new message the user recover the history of the chat with the last message. This behaviour I already see in other apps.

The other problem I notice is when the apk is compiled after send a message the chat changed with state CONNECTION FAILED. @KorbinianMossandl you notice this problem? Remember only compiled apk. Simulator or compile throught Android Studio works well.

**Comment by KorbinianMossandl** [comment]
i get a very unstable connection in general. but i think that is just zendesk. I added a reconnect button so the user can try to reconnect whenever it happens.

**Comment by rupas385** [comment]
yep, but only when I compile de APK? its very strange...

**Comment by KorbinianMossandl** [comment]
> yep, but only when I compile de APK? its very strange...

do you have different zendesk keys in your build settings based on type of build? like zendesk sandbox key when doing dev builds and normal key when doing prod builds? because sandbox gives me way more disconnects/unstable connection .

**Comment by KohlsAdrian** [comment]
you should reinitialize the providers, you should call `.dispose()` to release and `.startProviders()` to restart it, I won't release providers on `.endChat()` in native side, because is not the right way.
This is an expected behaviour

**Comment by rupas385** [comment]
> you should reinitialize the providers, you should call `.dispose()` to release and `.startProviders()` to restart it, I won't release providers on `.endChat()` in native side, because is not the right way.
> This is an expected behaviour

But why on iOS works fine and this problem only occurs in Android?

And why the providers starts automatically after the agent sent a message or start typing? If the startProviders() not called not have sense that the apps start works when agent start typing or sent message, not?

**Comment by KorbinianMossandl** [comment]
> you should reinitialize the providers, you should call `.dispose()` to release and `.startProviders()` to restart it, I won't release providers on `.endChat()` in native side, because is not the right way.
> This is an expected behaviour

you missunderstood. i do `.dispose()` and i do `.startProviders()` each time. but i dont do `.endChat()` when i want to keep the chat session alive. This is according to the native sdk documentation. And it also works if you store the last provder model and reload is.
the only issue is that you package does not send the last/current state when a new subscriber subscribes in this case.

**Comment by diegogarciar** [comment]
Hi, I think I'm having the same issue on iOS when I hot restart the application and the stream is reset. Anyone experiencing this?
